protocols/mdns/src/behaviour.rs

```diff
@@ -290,6 +296,11 @@ where
         &mut self,
         cx: &mut Context<'_>,
     ) -> Poll<ToSwarm<Self::ToSwarm, THandlerInEvent<Self>>> {
+        // If there are pending addresses to be emitted we emit them.
+        if let Some(event) = self.pending_events.pop_front() {
```

Comment:
why do you prefer having this here instead of after returning discovered addresses? 
I feel it's more correct to do this after returning the addresses and maintaining the previous flow of the `pool` loop.

Reply:
Thanks for your comments.

However, my idea here is that `Event::Discovered` and `ToSwarm::NewExternalAddrOfPeer` events are related. 

After `Event::Discovered` is returned in https://github.com/libp2p/rust-libp2p/blob/9a45db3f82b760c93099e66ec77a7a772d1f6cd3/protocols/mdns/src/behaviour.rs#L354, 
`poll` will return. 
When `poll` is entered next time, the `NewExternalAddrOfPeer` event will be returned because there is a `NewExternalAddrOfPeer` event in the queue.

The result is similar to this:
```
receive mdns::Event::Discovered Event
mDNS discovered a new peer: 12D3KooWGkhG2wR1mtadqShqiVEBjLHo8xk48ULrbkdftMsBbttR
NewExternalAddrOfPeer: /ip4/192.168.175.1/udp/63808/quic-v1/p2p/12D3KooWGkhG2wR1mtadqShqiVEBjLHo8xk48ULrbkdftMsBbttR, 12D3KooWGkhG2wR1mtadqShqiVEBjLHo8xk48ULrbkdftMsBbttR
```

Reply:
>However, my idea here is that Event::Discovered and ToSwarm::NewExternalAddrOfPeer events are related.

yeah exactly, that's why if we discover multiple addresses we shouldn't return **all** the `NewExternalAddrOfPeer` events before the second and subsequent `Event::Discovered(discovered)` right?

Reply:
Thank you for your detailed explanation. 
I have placed the return time of the `NewExternalAddrOfPeer` event after the `Event::Discovered` event. 
Please check whether it meets your expectations.